staging/vhost-device-gpu/src/main.rs

```diff
@@ -12,69 +12,118 @@
 pub mod gnu_main {
     use std::{path::PathBuf, process::exit};
 
-    use clap::Parser;
-    use log::{error, info};
-    use thiserror::Error as ThisError;
-    use vhost_device_gpu::{
-        device::{self, VhostUserGpuBackend},
-        GpuConfig, GpuMode,
-    };
-    use vhost_user_backend::VhostUserDaemon;
-    use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
-
-    type Result<T> = std::result::Result<T, Error>;
-
-    #[derive(Debug, ThisError)]
-    pub enum Error {
-        #[error("Could not create backend: {0}")]
-        CouldNotCreateBackend(device::Error),
-        #[error("Could not create daemon: {0}")]
-        CouldNotCreateDaemon(vhost_user_backend::Error),
-        #[error("Fatal error: {0}")]
-        ServeFailed(vhost_user_backend::Error),
+    use clap::{ArgAction, Parser, ValueEnum};
+    use log::error;
+    use vhost_device_gpu::{start_backend, GpuCapset, GpuConfig, GpuFlags, GpuMode};
+
+    #[derive(ValueEnum, Debug, Copy, Clone, Eq, PartialEq)]
+    #[repr(u64)]
+    pub enum CapsetName {
+        /// [virglrenderer] The virgl capset, superseded by Virgl2
+        Virgl = GpuCapset::VIRGL.bits(),
+        /// [virglrenderer] OpenGL implementation
+        Virgl2 = GpuCapset::VIRGL2.bits(),
+        /// [gfxstream] Vulkan implementation
+        GfxstreamVulkan = GpuCapset::GFXSTREAM_VULKAN.bits(),
+        /// [virglrenderer] Vulkan implementation
+        Venus = GpuCapset::VENUS.bits(),
+        /// [virglrenderer] unsuported/experimental: DRM native context support
+        Drm = GpuCapset::DRM.bits(),
+        /// [gfxstream] unsuported/experimental
+        GfxstreamMagma = GpuCapset::GFXSTREAM_MAGMA.bits(),
+        /// [gfxstream] unsuported/experimental: OpenGL ES implementation
+        GfxstreamGles = GpuCapset::GFXSTREAM_GLES.bits(),
+        /// [gfxstream] unsuported/experimental
+        GfxstreamComposer = GpuCapset::GFXSTREAM_COMPOSER.bits(),
+    }
```

Comment:
have you tested any of this unsupported capset? are they compatible with the vhost-device-gpu? lastly should`nt they be documented in the README? like what these capsets do?